File: backend/kube-tasks-backend/src/main/java/com/kube_tasks/kube_tasks_backend/services/TaskService.java
package com.kube_tasks.kube_tasks_backend.services;

import com.kube_tasks.kube_tasks_backend.dto.TaskRequestDTO;
import com.kube_tasks.kube_tasks_backend.dto.TaskResponseDTO;

import java.util.List;

public interface TaskService {
    List<TaskResponseDTO> getAllTasks();
    TaskResponseDTO getTaskById(Long id);
    TaskResponseDTO createTask(TaskRequestDTO taskRequestDTO);
    TaskResponseDTO updateTask(Long id, TaskRequestDTO taskRequestDTO);
    void deleteTask(Long id);
}

File: frontend/kube-tasks-front/src/app/services/task.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { TaskRequest, Task } from '../models/task.model';
import { environment } from '../../environments/environment.prod';


@Injectable({
  providedIn: 'root'
})
export class TaskService {
private apiUrl = environment.apiUrl;

  constructor(private http: HttpClient) { }

  getAllTasks(): Observable<Task[]> {
    return this.http.get<Task[]>(this.apiUrl);
  }

  getTaskById(id: number): Observable<Task> {
    return this.http.get<Task>(`${this.apiUrl}/${id}`);
  }

  createTask(task: TaskRequest): Observable<Task> {
    return this.http.post<Task>(this.apiUrl, task);
  }

  updateTask(id: number, task: TaskRequest): Observable<Task> {
    return this.http.put<Task>(`${this.apiUrl}/${id}`, task);
  }

  deleteTask(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}